AzureMonitorAgent/agent.py

```diff
@@ -0,0 +1,864 @@
+    return exit_code, output
+
+def enable():
+    """
+    Start the Azure Monitor Linux Agent Service
+    This call will return non-zero or throw an exception if
+    the settings provided are incomplete or incorrect.
+    Note: enable operation times out from WAAgent at 5 minutes
+    """
+    exit_if_vm_not_supported('Enable')
+
+    OneAgentEnableCommand = "systemctl start mdsd"
+
+    hutil_log_info('Handler initiating onboarding.')
+    exit_code, output = run_command_and_log(OneAgentEnableCommand)
+    return exit_code, output
+
+def disable():
+    """
+    Disable Azure Monitor Linux Agent process on the VM.
+    Note: disable operation times out from WAAgent at 15 minutes
+    """
+    #stop the Azure Monitor Linux Agent service
+    DisableOneAgentServiceCommand = "systemctl stop mdsd"
+
+    exit_code, output = run_command_and_log(DisableOneAgentServiceCommand)
+    return exit_code, output
+
+def update():
+    """
+    Update the current installation of AzureMonitorLinuxAgent
+    No logic to install the agent as agent -> install() will be called 
+    with udpate because upgradeMode = "UpgradeWithInstall" set in HandlerManifest
+    """
+    # print ("Uinstalling the current mdsd")
+    # exit_code, output = uninstall()
+    # if exit_code != 0:
+    #     return exit_code, output
+    # print ("installing the curretn mdsd")
+    # exit_code, output = install()
+    # print ("Update finished")
+    
+    return 0, ""
+
+
+
+# Dictionary of operations strings to methods
+operations = {'Disable' : disable,
+              'Uninstall' : uninstall,
+              'Install' : install,
+              'Enable' : enable,
+              'Update' : update,
+}
+
+
+def parse_context(operation):
+    """
+    Initialize a HandlerUtil object for this operation.
+    If the required modules have not been imported, this will return None.
+    """
+    hutil = None
+    if ('Utils.WAAgentUtil' in sys.modules
+            and 'Utils.HandlerUtil' in sys.modules):
+        try:
+            
+            logFileName = 'extension.log'
+            hutil = HUtil.HandlerUtility(waagent.Log, waagent.Error, logFileName=logFileName)
+            hutil.do_parse_context(operation)
+        # parse_context may throw KeyError if necessary JSON key is not
+        # present in settings
+        except KeyError as e:
+            waagent_log_error('Unable to parse context with error: ' \
+                              '{0}'.format(e))
+            raise ParameterMissingException
+    return hutil
+
+
+def find_package_manager(operation):
+    """
+    Checks if the dist is debian based or centos based and assigns the package manager accordingly
+    """
+    global PackageManager
+    global BundleFileName
+    dist, ver = find_vm_distro(operation)
+
+    dpkg_set = {"oracle", "debian", "ubuntu", "suse"}
+    rpm_set = {"redhat", "centos", "red hat"}
+    for dpkg_dist in dpkg_set:
+        if dist.lower().startswith(dpkg_dist):
```

Comment:
is startswith enough? also, break the for loop once the match is found

Reply:
Yes startswith is enough to test the azure distros, This is the way we currently use in omsagent installation as well. So they are well tested.